### PR TITLE
Use django-style logging support

### PR DESCRIPTION
**Pull request by eccles**
Problem:
The package defined its own root logger and this constrained
(unnecessarily) the useage by third parties.

Solution:
Added module-named loggers at each endpoint to allow ultimate
flexibility in configuring logging by the user.

Signed-off-by: Paul Hewlett <phewlett76@gmail.com>

### CODE REVIEW

File: archivist/archivist.py [hunk 1]
```diff
@@ -108,16 +111,6 @@ def __init__(self, url, *, auth=None, cert=None, verify=True):
...
-        self._assets = None
-        self._events = None
-        self._locations = None
-        self._attachments = None
-
-        self.assets: Optional[_AssetsClient]
-        self.events: Optional[_EventsClient]
-        self.locations: Optional[_LocationsClient]
-        self.attachments: Optional[_AttachmentsClient]
-
```
Comment (JAG-UK) [inline archivist/archivist.py:120]: These don't look like they're related to the logging change. Are they intentional/desirable?

Comment (eccles) [reply archivist/archivist.py:120]: the original statements setting the attributes to None is a mistake and I am intending to ask Serhiy whether he is OK with this...

Comment (eccles) [reply archivist/archivist.py:120]: otherwise _ will revert
_

Comment (JAG-UK) [reply archivist/archivist.py:120]: OK, fine then

**Review by JAG-UK — APPROVED** [review]

**Pull request merged**